Uploading development Python wheels to gemfury seems to be failing

I'm seeing:

```
  fury push \
    --api-token=${NANOARROW_GEMFURY_TOKEN} \
    --as="arrow-nightlies" \
    dist/*
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    NANOARROW_GEMFURY_TOKEN: ***
Top level ::CompositeIO is deprecated, require 'multipart/post' and use `Multipart::Post::CompositeReadIO` instead!
Top level ::Parts is deprecated, require 'multipart/post' and use `Multipart::Post::Parts` instead!
/opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/thor-1.0.1/lib/thor/error.rb:105:in '<class:Thor>': uninitialized constant DidYouMean::SPELL_CHECKERS (NameError)

    DidYouMean::SPELL_CHECKERS.merge!(
              ^^^^^^^^^^^^^^^^
Did you mean?  DidYouMean::SpellChecker
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/thor-1.0.1/lib/thor/error.rb:1:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/thor-1.0.1/lib/thor/base.rb:3:in 'Kernel#require_relative'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/thor-1.0.1/lib/thor/base.rb:3:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/thor-1.0.1/lib/thor.rb:2:in 'Kernel#require_relative'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/thor-1.0.1/lib/thor.rb:2:in '<top (required)>'
	from <internal:/opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/3.4.0/rubygems/core_ext/kernel_require.rb>:136:in 'Kernel#require'
	from <internal:/opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/3.4.0/rubygems/core_ext/kernel_require.rb>:136:in 'Kernel#require'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/gemfury-0.12.0/lib/gemfury/command.rb:5:in '<top (required)>'
	from <internal:/opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/3.4.0/rubygems/core_ext/kernel_require.rb>:136:in 'Kernel#require'
	from <internal:/opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/3.4.0/rubygems/core_ext/kernel_require.rb>:136:in 'Kernel#require'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/lib/ruby/gems/3.4.0/gems/gemfury-0.12.0/bin/fury:8:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/bin/fury:25:in 'Kernel#load'
	from /opt/hostedtoolcache/Ruby/3.4.1/x64/bin/fury:25:in '<main>'
Error: Process completed with exit code 1.
```

https://github.com/apache/arrow-nanoarrow/actions/runs/12552393397/job/34998992375#step:5:14

@raulcd has this been fixed elsewhere?

--- a/.github/workflows/python-wheels.yaml
+++ b/.github/workflows/python-wheels.yaml
@@ -174,8 +174,7 @@ jobs:
 
       - name: Install gemfury client
         run: |
-          # GH-36692: Pin gemfury due to wrong faraday dependency declaration.
-          gem install gemfury -v 0.12.0
+          gem install gemfury
 
       - name: Upload packages to Gemfury
         shell: bash

